openstack/blockstorage/v2/volumes/results.go

```diff
@@ -57,6 +77,25 @@ type Volume struct {
 	Multiattach bool `json:"multiattach"`
 }
 
+func (r *Volume) UnmarshalJSON(b []byte) error {
+	type tmp Volume
+	var s struct {
+		tmp
+		CreatedAt gophercloud.JSONRFC3339MilliNoZ `json:"created_at"`
```

Comment:
This is probably a silly question but why are we using gophercloud.JSONRFC3339MilliNoZ for CreatedAt and UpdatedAt?  As far as I can tell these fields are ISO8601 timestamps and so can contain a "Z". If we set the type to be JSONRFC3339MilliNoZ, json.Unmarshall will error out when processing a date such as "2017-01-04T16:09:33.860887324Z", which should, by my reading, of the specs should be valid.

There's no special marshalling for the CreatedAt and UpdatedAt fields in the images.Image structure yet the openstack API documentation for these fields seems to be identical to documentation for their counterparts in volumes.Volume.  Could we perhaps just delete this function and update the tags instead?